test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -528,6 +504,81 @@ def setUp(self):
         self.log_end_setup()
         self.measurements = []
 
+    @classmethod
+    def populate_test_list(cls):
+        cls.test_list = [attr for attr in dir(cls)
+                         if attr.startswith('test')
+                         and callable(getattr(cls, attr))]
+
+    @classmethod
+    def skip_cray_tests(cls):
+        if not cls.mom.is_cray():
+            return
+        msg = 'capability not supported on Cray'
+        # check if we need to skip the whole test suite
+        if '__skip_on_cray__' in cls.__dict__ \
+           and cls.__skip_on_cray__ is True:
+            for test_name in cls.test_list:
+                test_item = getattr(cls, test_name)
+                test_item.__unittest_skip__ = True
+                test_item.__unittest_skip_why__ = msg
+            return
```

Comment:
rather than return how out if/else? like
if check for class:
else:
   for each test